src/Sentry/Internal/MemoryMonitor.cs

```diff
@@ -0,0 +1,164 @@
+    {
+        _options = options;
+        _dumpTrigger = options.HeapDumpTrigger
+                       ?? throw new ArgumentException("No heap dump trigger configured on the options", nameof(options));
+        _onDumpCollected = onDumpCollected;
+        _onCaptureDump = onCaptureDump ?? CaptureMemoryDump;
+
+        _totalMemory = GC.GetGCMemoryInfo().TotalAvailableMemoryBytes;
+
+        GarbageCollectionMonitor.Start(CheckMemoryUsage, _cancellationTokenSource.Token);
+    }
+
+    internal void CheckMemoryUsage()
+    {
+        var eventTime = DateTimeOffset.UtcNow;
+        if (!_options.HeapDumpDebouncer.CanProcess(eventTime))
+        {
+            return;
+        }
+
+        var usedMemory = Environment.WorkingSet;
+        if (!_dumpTrigger(usedMemory, _totalMemory))
+        {
+            return;
+        }
+
+        _options.HeapDumpDebouncer.RecordOccurence(eventTime);
+
+        var usedMemoryPercentage = ((double)usedMemory / _totalMemory) * 100;
+        _options.LogDebug("Total Memory: {0:N0} bytes", _totalMemory);
+        _options.LogDebug("Memory used: {0:N0} bytes ({1:N2}%)", usedMemory, usedMemoryPercentage);
+        _options.LogDebug("Automatic heap dump triggered");
+        _onCaptureDump();
+    }
+
+    internal void CaptureMemoryDump()
+    {
+        if (_options.DisableFileWrite)
+        {
+            _options.LogDebug("File write has been disabled via the options. Unable to create memory dump.");
+            return;
+        }
+
+        var dumpFile = TryGetDumpLocation();
+        if (dumpFile is null)
+        {
+            return;
+        }
+
+        var processId = Environment.ProcessId;
+        _options.LogInfo("Creating a memory dump for Process ID: {0}", processId);
+
+        // Check which patth to use for dotnet-gcdump. If it's been bundled with the application, it will be available
+        // in the `dotnet-gcdump` folder of the application directory. Otherwise we assume it has been installed globally.
+        var bundledToolPath = Path.Combine(AppContext.BaseDirectory, "dotnet-gcdump", "dotnet-gcdump.dll");
+        if (File.Exists(bundledToolPath))
+        {
+            _options.LogDebug($"Using bundled version of dotnet-gcdump from: {bundledToolPath}");
+        }
+        else
+        {
+            _options.LogDebug("Using global version of dotnet-gcdump");
+        }
+
+        var arguments = $"collect -p {processId} -o '{dumpFile}'";
+        var command = File.Exists(bundledToolPath)
+            ? $"dotnet {bundledToolPath} {arguments}"
+            : $"dotnet-gcdump {arguments}";
+        var startInfo = new ProcessStartInfo
+        {
+            FileName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "cmd.exe" : "/bin/bash",
+            Arguments = $"-c \"{command}\"",
+            RedirectStandardOutput = true,
+            UseShellExecute = false,
+            CreateNoWindow = true,
```

Comment:
Don't we want this as `false`? This will have a popup show up, no?

Reply:
It's a confusing parameter - it's called Create**No**Window (not CreateWindow)... so setting this to true suppresses a popup.